### DISCUSSION TITLE
Action fails if it runs in a reusable workflow.

### DISCUSSION THREAD

**Discussion by sonomirco** [community/community#38256] [Actions] [answered]
Hi there, 
I have a reusable workflow used by a matrix.
`
 building:
    needs: bumping
    strategy:
      matrix: ${{ fromJSON(needs.bumping.outputs.versions) }}
      fail-fast: true
    uses: sandbox-2/TestNodebox/.github/workflows/deployment.yaml@main
`

If I run `azure/login@v1` inside the reusable workflow `deployment` I have this error.
`Error: Az CLI Login failed. Please check the credentials and make sure az is installed on the runner. For more information, refer https://aka.ms/create-secrets-for-GitHub-workflows`

It works perfectly if I use the action in a standard GitHub action, so the credential is not the problem.
The reusable workflow runs on `windows-latest`.
I tried the work around installing the cli before the login action using a powershell command, but I have the same error.
`

      - name: Install Azure cli
        shell: pwsh
        run: $ProgressPreference = 'SilentlyContinue'; Invoke-WebRequest -Uri https://aka.ms/installazurecliwindows -OutFile 
        .\AzureCLI.msi; Start-Process msiexec.exe -Wait -ArgumentList '/I AzureCLI.msi /quiet'; rm .\AzureCLI.msi
          
      - name: 🔑 Login via Az module
        uses: azure/login@v1
        with:
          creds: ${{secrets.AZURE_CREDS}}
          enable-AzPSSession: true
`

Do you have any idea how to solve the problem?

**Comment by airtower-luna** [comment, marked as answer]
Does the caller workflow pass the secret to the reusable workflow? It won't have access to the secret otherwise, and the login action will receive an empty string as `creds`.

See: https://docs.github.com/en/actions/using-workflows/reusing-workflows#using-inputs-and-secrets-in-a-reusable-workflow

**Reply by sonomirco** [reply]
> Does the caller workflow pass the secret to the reusable workflow? It won't have access to the secret otherwise, and the login action will receive an empty string as `creds`.
> 
> 
> 
> See: https://docs.github.com/en/actions/using-workflows/reusing-workflows#using-inputs-and-secrets-in-a-reusable-workflow

Yep!!
Thanks, I missed that part of the documentation 😅

**Reply by antlu** [reply]
Why does the called workflow itself can't read a secret from the repository where it resides? Is it a security concern?

**Reply by airtower-luna** [reply]
It's a general rule for reusable workflows. If the called workflow is in another repository someone could change e.g. it to send your secrets wherever. When caller and called workflow are in the same repository that's not really a problem. I assume applying the same rule is simply to avoid a complex special case (which would also break any calls from other repositories).
